src/act.movement.cpp

```diff
@@ -196,81 +197,99 @@ bool should_tch_see_chs_movement_message(struct char_data *tch, struct char_data
   if (tch == ch || PRF_FLAGGED(tch, PRF_MOVEGAG) || !AWAKE(tch) || IS_IGNORING(tch, is_blocking_ic_interaction_from, ch))
     return FALSE;
 
+  // If both tch and ch are NPCs, we don't care.
+  if (IS_NPC(tch) && IS_NPC(ch)) {
+    return FALSE;
+  }
+
   // Tuned out doing other things.
-  if (PLR_FLAGGED(tch, PLR_REMOTE) || PLR_FLAGGED(tch, PLR_MATRIX))
+  if (PLR_FLAGGED(tch, PLR_REMOTE) || PLR_FLAGGED(tch, PLR_MATRIX)) {
     return FALSE;
+  }
 
   // Failed to see from vehicle.
   if (tch->in_veh && (tch->in_veh->cspeed > SPEED_IDLE && success_test(GET_INT(tch), 4) <= 0)) {
     return FALSE;
   }
 
   // Check for stealth and other person-to-person modifiers.
-  if (we_care_about_sneaking && IS_AFFECTED(ch, AFF_SNEAK)) {
+  if (we_care_about_sneaking) {
```

Comment:
GitHub PR review isn't great for long changes. Am I correct in assuming that having the sneak check removed here means you can fail to spot someone you can see who's entering the room even if they're not sneaking? That doesn't seem like a desired outcome.

Reply:
That is correct. If ch is affected by invis, spirit concealment, or stealth/silence spells, and/or if tch is wounded, sustaining spells, non-invis vision penalties, or is otherwise distracted, then my thinking is that it should be possible for tch to fail to notice ch, even when ch is not actively sneaking.

If none of those apply, then 4 perception vs TN 2 is a 99.9% chance of success.

Reply:
Unfortunately I'm gonna get bug reports from both the hidden person and the non-hidden person from this behavior. "I got complained at by X for sneaking around when I wasn't" isn't a fun outcome for staff or players. Let's only have it where sneaking characters have a chance to go unseen when moving.

Reply:
There's another check below, at line 287 that comes into play. The "we_care_about_sneaking" flag is admittedly not well named, since it's actually "should we roll for perception". If NPC tch succeeds at a perception roll, there are 3 possible results depending on whether ch is doing something suspicious:
1. Ch is sneaking, then NPC tch becomes alert (as part of `should_tch_see_chs_movement_message`).
2, Ch is invis, then we do `process_spotted_invis` (as called by `do_simple_move`... should this happen inside `should_tch_see_chs_movement_message` instead?).
3. Ch is neither sneaking nor invis, so NPC doesn't react. This should more properly be "Ch is not suspicious", which may or may not include other spells and effects like spirit concealment and poltergeist, but that's a different discussion.